### ISSUE TITLE
Slugify headers

### ISSUE THREAD

**Issue by ArtskydJ** [TehShrike/noddity-template-parser#10]
It would be nice if headers were linkable.

At this level, we cannot guarantee unique IDs, since an embedded template might have the same header as its parent has.

> Original discussion: https://github.com/TehShrike/noddity/pull/33
> See also: https://github.com/jonschlinkert/remarkable/issues/122

We could extend remarkable. (If we end up making a markdown plugin, we can put it on npm.)

```js
var md = new Remarkable().use(function (remarkable) {
  remarkable.renderer.rules.heading_open = function (tokens, idx) {
    var hLevel = tokens[idx].hLevel
    var slug = slugify(tokens[idx + 1].content)
    return `<h${hLevel} id="${slug}">`
  }
})

function slugify(str) {
  return str.replace(/\W+/g, '-')
}
```

**Comment by TehShrike** [comment]
I think a Remarkable plugin would be the way to go.  I'm willing to deal with colliding ids as a separate issue (or, most likely, ignore it and let the browser do whatever it wants when there are duplicates).

**Comment by ArtskydJ** [comment]
As far as I can tell, no one has make a remarkable plugin to do this yet.

**Comment by ArtskydJ** [comment]
How to make a remarkable plugin: https://github.com/lemoinem/remarkable/blob/078d668ded82e52219e7284cba2fa3f39607d595/doc/plugins.md

**Comment by TehShrike** [comment]
Published as 1.1.0
